### PR TITLE
refactor: break everything

### PR DESCRIPTION
**Pull request by sshaman1101**
This PR starts as 'extract storage and reuse into worker',
but the number of dependencies into the state storage was too much.

Changed:
- State storage now lives in a separated package.
- State storage responsible only for things that must be saved on disk.
- Hub does not depend on storage, it is used only by Worker.
- Replace market with new blockchain-backed API.
- Stub for DWH API.
- ETH wrapper for the Hub was replaced with MarketAPI.
- ACLs for workers are removed.
- Hub <-> Worker handshake removed.
- DealMeta and TaskMeta are moved into the `structs` package.
- Drop meaningless Hub's methods (device properties, miner management, ACLs).
- Deals and ask-plans now handled on Worker's side.
- ProposeDeal and ApproveDeal are deprecated.
- Everything useless removed from config files.

### CODE REVIEW

**Comment by 3Hren** [comment]
> ACLs for workers are removed.
Why? When the master-slave keys be introduced it'll be necessary to return this back.

File: insonmnia/miner/server.go [hunk 1]
```diff
@@ -255,40 +276,7 @@ func (m *Miner) deleteTaskMapping(id string) {
 	delete(m.nameMapping, id)
 }
 
-// Ping works as Healthcheck for the Hub
-func (m *Miner) Ping(ctx context.Context, _ *pb.Empty) (*pb.PingReply, error) {
-	log.G(m.ctx).Info("got ping request from Hub")
-	return &pb.PingReply{}, nil
-}
-
-// Info returns runtime statistics collected from all containers working on this miner.
```
Comment (3Hren) [inline insonmnia/miner/server.go]: Is this irrelevant now? If not - why to delete?

File: insonmnia/miner/server.go [hunk 2]
```diff
@@ -404,10 +392,12 @@ func (m *Miner) Start(ctx context.Context, request *pb.MinerStartRequest) (*pb.M
 		return nil, status.Errorf(codes.Unauthenticated, "invalid public key provided %v", err)
 	}
 
+	// fixme: this
```
Comment (3Hren) [inline insonmnia/miner/server.go]: Fix what?

File: insonmnia/miner/server.go [hunk 3]
```diff
@@ -937,6 +926,97 @@ func getDescriptionForBenchmark(b *pb.Benchmark) Description {
...
+
+func (m *Miner) CreateAskPlan(ctx context.Context, request *pb.CreateAskPlanRequest) (string, error) {
+	log.G(m.ctx).Info("handling CreateAskPlan request", zap.Any("request", request))
+
+	ord := &pb.Order{
+		OrderType:      pb.OrderType_ASK,
+		Slot:           request.GetSlot(),
+		ByuerID:        request.BuyerID,
+		PricePerSecond: request.PricePerSecond,
+		SupplierID:     util.PubKeyToAddr(m.ethkey.PublicKey).Hex(),
```
Comment (3Hren) [inline insonmnia/miner/server.go]: Stop using ` util.PubKeyToAddr` please.

File: insonmnia/state/state.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+type Storage struct {
+	mu  sync.Mutex
+	ctx context.Context
+
+	store store.Store
+	data  *stateJSON
+}
+
+type StorageConfig struct {
+	Endpoint string `required:"true" default:"/var/lib/sonm/worker.boltdb" yaml:"endpoint"`
```
Comment (3Hren) [inline insonmnia/state/state.go]: `yaml` first.

File: insonmnia/state/state.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+func NewState(ctx context.Context, cfg *StorageConfig) (*Storage, error) {
+	clusterStore, err := makeStore(ctx, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	out := &Storage{
+		ctx:   ctx,
+		store: clusterStore,
+		data:  newEmptyState(),
```
Comment (3Hren) [inline insonmnia/state/state.go]: `newEmptyState()` is called twice: here and during `loadInitial`

File: insonmnia/structs/task.go [hunk 6]
```diff
@@ -32,6 +33,20 @@ func (r *StartTaskRequest) GetDeal() *sonm.Deal {
 	return r.Deal
 }
 
-func (r *StartTaskRequest) GetDealId() string {
-	return r.GetDeal().GetId()
+func (r *StartTaskRequest) GetDealId() *sonm.BigInt {
+	v, _ := sonm.NewBigIntFromString(r.GetDeal().GetId())
```
Comment (3Hren) [inline insonmnia/structs/task.go]: So this can't error? Let's bet?

File: insonmnia/state/state.go [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+	_, ok := s.data.AskPlans[planID]
+	if !ok {
+		return errors.New("specified ask-plan does not exist")
+	}
+
+	delete(s.data.AskPlans, planID)
+	return s.dump()
+}
+
+func (s *Storage) GetPassedBenchmarks() map[uint64]bool {
```
Comment (3Hren) [inline insonmnia/state/state.go]: No `Get` prefixes please.

File: insonmnia/node/tasks.go [hunk 8]
```diff
@@ -57,16 +57,16 @@ func (t *tasksAPI) List(ctx context.Context, req *pb.TaskListRequest) (*pb.TaskL
...
 		t.getSupplierTasks(ctx, tasks, deal)
 	}
 
 	return &pb.TaskListReply{Info: tasks}, nil
 }
 
-func (t *tasksAPI) getSupplierTasks(ctx context.Context, tasks map[string]*pb.TaskListReply_TaskInfo, deal *pb.Deal) {
-	hub, cc, err := t.getHubClientByEthAddr(ctx, deal.GetSupplierID())
+func (t *tasksAPI) getSupplierTasks(ctx context.Context, tasks map[string]*pb.TaskStatusReply, deal *pb.Deal) {
+	hub, cc, err := getHubClientByEthAddr(ctx, t.remotes, deal.GetSupplierID())
```
Comment (3Hren) [inline insonmnia/node/tasks.go]: Why?

File: blockchain/api.go [hunk 9]
```diff
@@ -7,17 +7,19 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
```
Comment (3Hren) [inline blockchain/api.go]: Put these changes for blockchain API in a separate PR.

File: etc/hub.yaml [hunk 10]
```diff
@@ -2,12 +2,6 @@
 # allowed format is "ip:port" to bind Hub to given IP address
 # or ":port" to bind Hub to all available IPs
 endpoint: ":15010"
-
```
Comment (3Hren) [inline etc/hub.yaml]: Seems like a candidate for a separate PR.

File: insonmnia/hub/server.go [hunk 11]
```diff
@@ -391,26 +312,12 @@ func (h *Hub) startTask(ctx context.Context, request *structs.StartTaskRequest)
 		return nil, status.Errorf(codes.Internal, "failed to start %v", err)
 	}
 
-	info := TaskInfo{*request, *response, taskID, dealID, "self", nil}
-
-	err = h.state.SaveTask(DealID(request.GetDealId()), &info)
```
[the remark this answers is not in the document]
Comment (sshaman1101) [reply insonmnia/hub/server.go]: We keep statuses for all tasks into a Worker instance.

File: insonmnia/hub/miner_ctx.go [hunk 12]
```diff
@@ -165,34 +143,29 @@ func (m *MinerCtx) releaseDeal(id OrderID) {
...
 	defer m.mu.Unlock()
 	return m.orderUsage(id)
 }
 
-func (m *MinerCtx) orderUsage(id OrderID) (resource.Resources, error) {
+func (m *MinerCtx) orderUsage(id structs.OrderID) (resource.Resources, error) {
 	usage, exists := m.usageMapping[id]
 	if !exists {
-		return resource.Resources{}, errOrderNotExists
+		return resource.Resources{}, fmt.Errorf("order with id=%s does not found", id)
```
Comment (3Hren) [inline insonmnia/hub/miner_ctx.go]: Grammar

File: insonmnia/hardware/marshal.go [hunk 13]
```diff
@@ -1,31 +1,40 @@
...
 
-func MemoryIntoProto(m *mem.Device) *sonm.RAMDevice {
-	return &sonm.RAMDevice{
-		Total: m.Total,
+	return &sonm.DevicesReply{
+		CPUs:    cpus,
+		GPUs:    gpus,
+		Memory:  ram,
+		Network: net,
+		Storage: stor,
```
Comment (3Hren) [inline insonmnia/hardware/marshal.go]: Giving a variable name with more than 4 characters awakes a long-sleeping memories about cruel of 60x?

**Pull request merged**